NitroxClient/GameLogic/Spawning/Metadata/EntityMetadataManager.cs

```diff
@@ -43,7 +43,11 @@ public Optional<EntityMetadata> Extract(GameObject o)
         {
             if (extractors.TryGetValue(component.GetType(), out IEntityMetadataExtractor extractor))
             {
-                return extractor.From(component);
+                Optional<EntityMetadata> metadata = extractor.From(component);
+                if (metadata.HasValue)
+                {
+                    return metadata;
+                }
```

Comment:
Sure, this is a consequence of how bad EntityMetadata currently is and it'll surely be reworked in the future so that one entity can hold multiple ones. It happened that I introduced for the first time a metadata (EatableMetadata) that can be in conflict with another one (WaterParkCreatureMetadata). In this case, WaterParkCreatureMetadata will be found first so we don't want to return it if it's null (it's possible that the extractor returns null) and instead we'll let it run until it finds a valid one.